### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by ilyazzz**
There are still some things I would like to have before merging this:

- [X] Create documentation on how to use it
- [] Push images to docker hub

### CODE REVIEW

**Comment by NotNotQuinn** [comment]
Tried running, couldn't get docker to use local `supidb` image, unless im doing something wrong.
Also no idea how to get the redis thing that docker-compose.yml mentions. (no dockerfile to make it)

Looking forward to this being a thing.

**Comment by ilyazzz** [comment]
@NotNotQuinn to build the images locally, you need to do:
`docker build . -t supibot -f Dockerfile`
`docker build . -t supidb -f Dockerfile-db`
Eventually I plan on uploading the images to docker hub so this won't be needed.

Redis gets downloaded automatically when the container is created.

File: docs/setup.md [hunk 1]
```diff
@@ -1,28 +1,70 @@
...
+      restart: unless-stopped
+      environment:
+        - "MARIA_HOST=db"
+        - "MARIA_USER=supibot"
+        - "MARIA_PASSWORD=supibot"
+        - "REDIS_CONFIGURATION=redis"
+        - "COMMAND_PREFIX=|"
+        - "INITIAL_BOT_NAME=supibot"
+        - "INITIAL_PLATFORM=twitch"
+        - "INITIAL_CHANNEL=supinic"
```
Comment (Supinic) [inline docs/setup.md:34]: these values should use placeholders like the client id and oauth

File: docs/setup.md [hunk 2]
```diff
@@ -1,28 +1,70 @@
...
+        - "TWITCH_CLIENT_ID=<your_client_id>"
+        - "TWITCH_OAUTH=<your_oauth>"
```
Comment (Supinic) [inline docs/setup.md:36]: would it make sense to also include other platforms' oauths here? along with platform being one of the ones we support

File: init/setup.js [hunk 3]
```diff
@@ -70,11 +70,13 @@
 	}
 	console.log("Database credentials setup successfully.");
 
-	let packageManager = null;
```
Comment (Supinic) [inline init/setup.js:73]: the entire file should be reverted, as we use `auto-setup` for working with env values

**Review by Supinic — CHANGES_REQUESTED** [review]

File: init/setup.js (hunk 3 above)
Comment (ilyazzz) [reply init/setup.js]: done

**Review by Supinic — APPROVED** [review]

**Pull request merged**